### PR TITLE
Fixing deletion errors in text matrix parsing code

### PR DESCRIPTION
**Pull request by danijel3**
Fixing the errors mentioned in issue #1119

### CODE REVIEW

File: src/matrix/kaldi-matrix.cc [hunk 1]
```diff
@@ -1391,12 +1392,13 @@ void Matrix<Real>::Read(std::istream & is, bool binary, bool add) {
...
               specific_error << "Matrix has inconsistent #cols: " << num_cols
                              << " vs." << data[i]->size() << " (processing row"
-                             << i;
+                             << i << ")";
               goto cleanup;
             }
             for (int32 j = 0; j < num_cols; j++)
               (*this)(i, j) = (*(data[i]))[j];
             delete data[i];
+            data[i]=NULL;
```
Comment (danijel3) [inline src/matrix/kaldi-matrix.cc]: Sure. Is it okay now?

**Comment by danpovey** [comment]
yes but I'll wait for automatic checks to complete and then merge.  Thanks!

File: src/matrix/kaldi-matrix.cc [hunk 2]
```diff
@@ -1437,9 +1439,11 @@ void Matrix<Real>::Read(std::istream & is, bool binary, bool add) {
     // Note, we never leave the while () loop before this
     // line (we return from it.)
  cleanup: // We only reach here in case of error in the while loop above.
-    delete cur_row;
+    if(cur_row != NULL)
```
Comment (vdp) [inline src/matrix/kaldi-matrix.cc]: Just FYI, the check for NULL is not necessary (http://stackoverflow.com/questions/4190703/is-it-safe-to-delete-a-null-pointer).

Comment (danijel3) [inline src/matrix/kaldi-matrix.cc]: Okay, you're right. I may be overcompensating a bit, cause I treat Kaldi as something that is supposed to outlive many changes in standards and compiler versions.

**Pull request merged**